internal/internal_task_handlers.go

```diff
@@ -1835,7 +1845,11 @@ func traceLog(fn func()) {
 	}
 }
 
-func workflowCategorizedByTimeout(executionTimeout int32) string {
+func workflowCategorizedByTimeout(wfContext *workflowExecutionContextImpl) string {
+	if wfContext == nil || wfContext.workflowInfo == nil {
+		return "unknown"
+	}
```

Comment:
nils are not possible here (as it's currently used), we can/should drop the checks.

Reply:
nils are possible in this commit due to where this function is called. I checked with @timl3136 about our intent for `workflowruntimelength`. We will only have it for cache hit metric for now. So moved the scope tagging to appropriate if block back.

Reply:
ooh, I see, because it was moved outside that != nil check.  yea, good point.

is this tag relevant for the cache miss branch further down too?  I'm kinda feeling like "no" but not totally confident.

Reply:
we don't have the workflow context for misses in this path so not applicable for it.